src/ast/visitor.ts

```diff
@@ -0,0 +1,283 @@
+// Copyright (c) 2021 the Skulpt Project
+// SPDX-License-Identifier: MIT
+
+/* File automatically generated by gen_asdl/__main__.py. */
+// deno-lint-ignore-file no-explicit-any
+
+import type {
+    AST,
+    Module,
+    Interactive,
+    Expression,
+    FunctionType,
+    FunctionDef,
+    AsyncFunctionDef,
+    ClassDef,
+    Return,
+    Delete,
+    Assign,
+    AugAssign,
+    AnnAssign,
+    For,
+    AsyncFor,
+    While,
+    If,
+    With,
+    AsyncWith,
+    Raise,
+    Try,
+    Assert,
+    Import,
+    ImportFrom,
+    Global,
+    Nonlocal,
+    Expr,
+    Pass,
+    Break,
+    Continue,
+    Debugger,
+    BoolOp,
+    NamedExpr,
+    BinOp,
+    UnaryOp,
+    Lambda,
+    IfExp,
+    Dict,
+    Set_,
+    ListComp,
+    SetComp,
+    DictComp,
+    GeneratorExp,
+    Await,
+    Yield,
+    YieldFrom,
+    Compare,
+    Call,
+    FormattedValue,
+    JoinedStr,
+    Constant,
+    Attribute,
+    Subscript,
+    Starred,
+    Name,
+    List,
+    Tuple,
+    Slice,
+    comprehension,
+    ExceptHandler,
+    arguments_,
+    arg,
+    keyword,
+    alias,
+    withitem,
+    TypeIgnore,
+} from "./astnodes.ts";
+
+export abstract class ASTVisitor {
+    visitSeq(seq: AST[] | null) {
+        if (seq === null) return null;
+        for (const node of seq) {
+            node.walkabout(this);
+        }
+    }
+
+    defaultVisitor(_node: AST): any {
+        throw new Error("NodeVisitor not implemented");
+    }
+
+    visit_Module(node: Module): any {
```

Comment:
Why dis you go with the `_` we now have snake_Camel case. 😂

Reply:
I started with just joining the names but I really didn't like 

`visitarguments_` or `visitalias`

And then I thought about capitalising but we have `expr` and `Expr` nodes. 

So then I just went with what pypy had...?

Reply:
Maybe we could change the names of some of the nodes...

what if we kept abstract AST classes lower case, and then actual classes capitalized.
i.e. `stmt` remains `stmt`
but
`alias` => `Alias`. 
`arguments_` => `Arguments`
Think it affects 6 classes also `comprehension`, `keyword`, `withitem`, `arg`

Reply:
Ah ok I understand lets not change the class names.